kernel/kernel-auth-adapter/src/main/java/io/mosip/kernel/auth/defaultadapter/config/ClientTokenRestInterceptor.java

```diff
@@ -0,0 +1,150 @@
+				AuthAdapterConstant.AUTH_HEADER + memoryCache.get(CLIENT_TOKEN_KEY));
+		return execution.execute(request, body);
+	}
+
+	private String getClientToken(String clientID, String clienSecret, String appID) {
+		RequestWrapper<ClientSecret> requestWrapper = new RequestWrapper<>();
+		ClientSecret clientCred = new ClientSecret();
+		clientCred.setAppId(appID);
+		clientCred.setClientId(clientID);
+		clientCred.setSecretKey(clienSecret);
+		requestWrapper.setRequest(clientCred);
+		HttpEntity<String> response = null;
+		try {
+			response = restTemplate.postForEntity(tokenURL, requestWrapper, String.class);
+		} catch (HttpServerErrorException | HttpClientErrorException e) {
+			LOGGER.error("error connecting to auth service {}", e.getResponseBodyAsString());
+			throw new AuthAdapterException(AuthAdapterErrorCode.CANNOT_CONNECT_TO_AUTH_SERVICE.getErrorCode(),
+					e.getResponseBodyAsString());
+		}
+		if (response == null) {
+			throw new AuthAdapterException(AuthAdapterErrorCode.CANNOT_CONNECT_TO_AUTH_SERVICE.getErrorCode(),
+					AuthAdapterErrorCode.CANNOT_CONNECT_TO_AUTH_SERVICE.getErrorMessage());
+		}
+		String responseBody = response.getBody();
+		List<ServiceError> validationErrorList = ExceptionUtils.getServiceErrorList(responseBody);
+		if (!validationErrorList.isEmpty()) {
+			throw new AuthRestException(validationErrorList);
+		}
+		HttpHeaders headers = response.getHeaders();
+		List<String> cookies = headers.get("Set-Cookie");
+		if(cookies==null|| cookies.isEmpty())
+			throw new AuthAdapterException(AuthAdapterErrorCode.IO_EXCEPTION.getErrorCode(),
+					AuthAdapterErrorCode.IO_EXCEPTION.getErrorMessage());
+		
+		String authToken = cookies.get(0).split(";")[0].split(AuthAdapterConstant.AUTH_HEADER)[1];
+
+		return authToken;
+	}
+
+	private boolean isTokenValid(String authToken) {
+		HttpHeaders headers = new HttpHeaders();
+		headers.add(AuthAdapterConstant.COOKIE, AuthAdapterConstant.AUTH_HEADER + authToken);
+		HttpEntity<String> requestEntity = new HttpEntity<>(headers);
+		HttpEntity<String> response = null;
+		try {
+			response = restTemplate.exchange(validateTokenURL, HttpMethod.GET, requestEntity, String.class);
+		} catch (HttpServerErrorException | HttpClientErrorException e) {
+			if (e.getStatusCode() == HttpStatus.UNAUTHORIZED) {
+				LOGGER.error("Token not valid {}", e.getResponseBodyAsString());
+				return false;
+			}
+			throw new AuthAdapterException(AuthAdapterErrorCode.CANNOT_CONNECT_TO_AUTH_SERVICE.getErrorCode(),
+					e.getResponseBodyAsString());
+		}
+		if (response == null) {
+			throw new AuthAdapterException(AuthAdapterErrorCode.CANNOT_CONNECT_TO_AUTH_SERVICE.getErrorCode(),
+					AuthAdapterErrorCode.CANNOT_CONNECT_TO_AUTH_SERVICE.getErrorMessage());
+		}
+		String responseBody = response.getBody();
+		List<ServiceError> validationErrorList = ExceptionUtils.getServiceErrorList(responseBody);
+		if (!validationErrorList.isEmpty()) {
+			throw new AuthRestException(validationErrorList);
```

Comment:
Why do we need 2 differnt exception classes AuthAdapterException vs AuthRestException ?

Reply:
AuthAdapter Exception handles single exception, while auth RestException handles list of exception got after a rest call by authServer